### PR TITLE
feat(core): hook-up nimma v2

### LINKED ISSUE #1260: Nimma usage generates very different results
**Describe the bug**
5.5.0 introduces an optional rule jsonpath optimizer: nimma. It seems to not properly parse some paths.

**To Reproduce**

**ruleset:**
```yaml
rules:
  read-scope-prefix-on-get:
    type: validation
    severity: warn
    recommended: true
    formats: ["oas3"]
    description: Scope should start with something
    message: Scope should start with something
    given: "$.paths.*[?(@property === 'get' || @property === 'options')].security.*.my_auth.*"
    then:
      function: pattern
      functionOptions:
        match: "^urn:"

  no-404-response:
    type: validation
    severity: info
    recommended: true
    formats: ["oas3"]
    description: Operation must not define a 404 response when no ressource identifier.
    message: The request must not define a 404 response.
    given: "$.paths[?(!@property.includes('_id}'))][?(@property === 'get' || @property === 'post' || @property === 'delete' || @property === 'options' || @property === 'patch' )].responses"
    then:
      field: "404"
      function: falsy
```

**document:**
```yaml
openapi: 3.0.0

paths:
  /any/resources/{resource}:
    get:
      responses:
        "200":
          description: All is good
        "404":
          description: All is nowhere"
  /resources:
    get: &dummy-endpoint
      responses:
        "200":
          description: All is good
      security:
        - my_auth:
            - urn:something
    patch:
      <<: *dummy-endpoint
      security:
        - my_auth:
            - Urn:wrong-but-should-go-unnoticed
```
**without nimma:**
```bash
$ node -r ts-node/register -r tsconfig-paths/register src/cli/index.ts lint -r ./repro/nimma.ruleset.yaml ./repro/nimma.openapi.yaml
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped
OpenAPI 3.x detected

c:/spectral/repro/nimma.openapi.yaml
 9:15  information  no-404-response  The request must not define a 404 response.

✖ 1 problem (0 errors, 0 warnings, 1 info, 0 hints)

```

**activating nimma:**
```bash
$ node -r ts-node/register -r tsconfig-paths/register src/cli/index.ts lint -r ./repro/nimma.ruleset.yaml ./repro/nimma.openapi.yaml
Missing baseUrl in compilerOptions. tsconfig-paths will be skipped
OpenAPI 3.x detected

c:/_work/spectral/repro/nimma.openapi.yaml
 23:15  warning  read-scope-prefix-on-get  Scope should start with something

✖ 1 problem (0 errors, 1 warning, 0 infos, 0 hints)
```

**Expected behavior**
Using nimma or not should not change the return results.

**Environment (remove any that are not applicable):**
 - Library version: develop branch
 - OS: Win10
 - Browser: N/A

### LINKED ISSUE #1456: Spectral not handling keys containing @
This refactor will help us to allow below schema:
'@customerLocation':
 type: string
 example: "https://google.com"
 format: uri
 description: A URI to a JSON-Schema file that defines additional attributes and relationships
 
 As of now, we are getting below exception with this-
 TypeError: Unknown value type customerTy
    at JSONPath._trace (/*/node_modules/jsonpath-plus/dist/index-umd.js:766:17)
    at /*/node_modules/jsonpath-plus/dist/index-umd.js:745:23
    at /*/node_modules/jsonpath-plus/dist/index-umd.js:909:9
    at Array.forEach (<anonymous>)
    at JSONPath._walk (/*/node_modules/jsonpath-plus/dist/index-umd.js:908:24)
    at JSONPath._trace (/*/node_modules/jsonpath-plus/dist/index-umd.js:743:12)
    at JSONPath._trace (/*/node_modules/jsonpath-plus/dist/index-umd.js:698:19)
    at /*/node_modules/jsonpath-plus/dist/index-umd.js:706:23
    at /*/node_modules/jsonpath-plus/dist/index-umd.js:909:9
    at Array.forEach (<anonymous>)

### PR DESCRIPTION
**Pull request by P0lip**
Fixes #1260  
Fixes https://github.com/stoplightio/spectral/issues/1456

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Additional context**

Here's the source code https://github.com/P0lip/nimma/commit/4dd6daa149fe458ef5fe41e3c6885cf33c6c4010

The following PR adds Nimma back - for eternity this time around as the new version of it by far more spec-compliant than the former version, and actually does have a number of [test suites](https://github.com/P0lip/nimma/blob/master/src/__tests__/compatibility.test.mjs) comparing the results between the libraries.
As a result of that I'm more confident about the results, and thus we should be safely roll it out to everyone, not just the ones that have the ENV variable set or passed the useNimma option.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -36,10 +37,11 @@
     "blueimp-md5": "2.18.0",
     "expression-eval": "4.0.0",
     "json-schema": "0.3.0",
-    "jsonpath-plus": "5.0.7",
+    "jsonpath-plus": "6.0.1",
     "lodash": "~4.17.21",
+    "lodash.topath": "^4.5.2",
     "minimatch": "3.0.4",
-    "nimma": "0.0.0",
+    "nimma": "p0lip/nimma#2bbca1be53a9fe63bdbb5d7ff89e175646897db3",
```
Comment (P0lip) [inline packages/core/package.json:44]: todo before the merge: publish the package & ideally add some docs

**Comment by billiegoose** [comment]
Does it work?

**Comment by billiegoose** [comment]
I guess what I really mean is... Have you got a PR where it's yalced in Platform we can play with?

**Review by mnaumanali94 — APPROVED** [review]
Lets get this inn!

**Pull request merged**